cmake/ElektraCompiling.cmake

```diff
@@ -61,10 +61,27 @@ if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
 
 	if (ENABLE_DEBUG)
 		set (EXTRA_FLAGS "${EXTRA_FLAGS} -fsanitize=undefined -fsanitize=integer")
-		set (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -fsanitize=undefined -fsanitize=integer")
+		set (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -lubsan")
 	endif()
 endif()
 
+if (ENABLE_ASAN)
+	set (EXTRA_FLAGS "${EXTRA_FLAGS} -fsanitize=address -fno-omit-frame-pointer")
+	set (ASAN_LIBRARY "-lasan") #this is needed for GIR to put asan in front
+	if (CMAKE_COMPILER_IS_GNUCXX)
+		set (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -fsanitize=address")
+		# this is currently a workaround so gtests compile with ASAN
+		#TODO someone with expertise migrate this into LibAddTest
+		set (COMMON_FLAGS "${COMMON_FLAGS} -lpthread")
```

Comment:
> find_package(Threads)
> and then you can add ${CMAKE_THREAD_LIBS_INIT} to linker flags

This is already used in LibAddTest and i still run into compiling issues. Adding it globally fixed theme. Those the question if someone with expertise in this can solve this better.